Sources/StreamVideoSwiftUI/AnyViewFactory.swift

```diff
@@ -0,0 +1,207 @@
+//
+// Copyright © 2025 Stream.io Inc. All rights reserved.
+//
+
+import StreamVideo
+import StreamWebRTC
+import SwiftUI
+
+public final class AnyViewFactory: ViewFactory {
+
+    private let _makeCallControlsView: (CallViewModel) -> AnyView
+    private let _makeOutgoingCallView: (CallViewModel) -> AnyView
+    private let _makeJoiningCallView: (CallViewModel) -> AnyView
+    private let _makeIncomingCallView: (CallViewModel, IncomingCall) -> AnyView
+    private let _makeWaitingLocalUserView: (CallViewModel) -> AnyView
+    private let _makeVideoParticipantsView: (
+        CallViewModel,
+        CGRect,
+        @escaping @MainActor(CallParticipant, Bool) -> Void
+    ) -> AnyView
+    private let _makeVideoParticipantView: (
+        CallParticipant,
+        String,
+        CGRect,
+        UIView.ContentMode,
+        [String: RawJSON],
+        Call?
+    ) -> AnyView
+    #if compiler(>=6.0)
+    private let _makeVideoCallParticipantModifier: (
+        CallParticipant,
+        Call?,
+        CGRect,
+        CGFloat,
+        Bool
+    ) -> any ViewModifier
+    #endif // any ViewModifier not recognised on Swift version < 6.0
```

Comment:
Everything here would be hard to maintain - I think we're better off by just passing the default concrete type.

Reply:
I'm not going to argue that but i think that the Generic requirement our Views have is also quite limited and causing issues on reusability. We can think about it and decide in the future what to do with it.